packages/slate-editor/src/extensions/variables/useVariables.ts

```diff
@@ -16,11 +19,31 @@ function placeholderToOption(placeholder: Variable): Option<Variable> {
 
 const DEFAULT_PARAMETERS: VariablesExtensionParameters = { variables: [] };
 
-export function useVariables({ variables }: VariablesExtensionParameters = DEFAULT_PARAMETERS) {
+export function useVariables(
+    editor: Editor,
+    { variables }: VariablesExtensionParameters = DEFAULT_PARAMETERS,
+) {
     const options = useMemo(() => variables.map(placeholderToOption), [variables]);
+    const isEnabled = useCallback(
+        (range: BaseRange | null) => {
+            if (!range) {
+                return true;
+            }
+
+            const nodes = Array.from(
+                Editor.nodes(editor, {
+                    at: range,
+                    match: (node) => isTitleHeadingNode(node) || isSubtitleHeadingNode(node),
+                }),
+            );
+            return nodes.length === 0;
+        },
+        [editor],
+    );
```

Comment:
We do not have any mechanism to disallow mentions functionality as far as I'm aware so I had to came up with something like this.

For now, it disables mentions for title and subtitle but we should consider checking other nodes as well. For example, it's possible to use placeholders in image caption but the story is not able to render it.